exercise-test-doubles-5-95/test/🏅your-mission-test-5-95.js

```diff
@@ -42,72 +45,150 @@ test("When video instructions are valid, then a success email should be sent to
-// });
-
-// test("When video instructions are valid, then a success email should be sent to creator (step on toe2)", async () => {
-//   // Arrange
-//   const clipInstructions = testHelper.factorClipInstructions({
-//     creator: { email: "yoni@testjavascript.com", name: "Yoni" },
-//     destination: "Mexico",
-//   });
-//   const tripClipServiceUnderTest = new TripClipService();
-//   const spyOnMailer = sinon.stub(mailSender, "send");
-
-//   // Act
-//   await tripClipServiceUnderTest.generateClip(clipInstructions);
-
-//   // Assert
-//   expect(spyOnMailer.lastCall.args).toEqual(["yoni@testjavascript.com", expect.any(String)]);
-// });
+test("When the video production fails, then no email is sent (step on toe1)", async () => {
+  // Arrange
+  const clipInstructions = testHelper.factorClipInstructions({
+    creator: { email: "yoni@testjavascript.com", name: "Yoni" },
+    destination: "Mexico",
+  });
+  const tripClipServiceUnderTest = new TripClipService();
+  sinon.stub(videoProducer, "produce").rejects(new Error("I just failed "));
+  const spyOnMailer = sinon.stub(mailSender, "send");
+
+  // Act
+  try {
+    await tripClipServiceUnderTest.generateClip(clipInstructions);
+  } catch (e) {
+    //We don't care about the error here
+  }
+
+  // Assert
+  expect(spyOnMailer.called).toBe(false);
+});
+
+test("When video instructions are valid, then a success email should be sent to creator (step on toe2)", async () => {
+  // Arrange
+  const clipInstructions = testHelper.factorClipInstructions({
+    creator: { email: "yoni@testjavascript.com", name: "Yoni" },
+    destination: "Mexico",
+  });
+  const tripClipServiceUnderTest = new TripClipService();
+  const spyOnMailer = sinon.stub(mailSender, "send");
+
+  // Act
+  await tripClipServiceUnderTest.generateClip(clipInstructions);
+
+  // Assert
+  expect(spyOnMailer.lastCall.args).toEqual([
+    "yoni@testjavascript.com",
+    expect.any(String),
+  ]);
+});
 
 // ✅ TASK: Test that when the VideoProducer.produce operation operation fails, an exception is thrown
 // with a property name: 'video-production-failed'
 // 💡 TIP: Use a test double that can change the response of this function and trigger it to throw an error
 // 💡 TIP: This is grey box testing, we mess with the internals but with motivation to test the OUTCOME of the box
-
+test("When the VideoProducer.produce operation operation fails, then an exception is thrown", async () => {
+  //Arrange
+  const clipInstructions = testHelper.factorClipInstructions({
+    creator: { email: "yoni@testjavascript.com", name: "Yoni" },
+    destination: "Mexico",
+  });
+  const tripClipServiceUnderTest = new TripClipService();
+  sinon.stub(videoProducer).produce.rejects(new Error("I just failed "));
+  // Act
+  const generateClipMethodWrapper = async () => {
+    await tripClipServiceUnderTest.generateClip(clipInstructions);
+  };
+  // Assert
+  expect(generateClipMethodWrapper).rejects.toThrowError(
```

Comment:
I'm pretty sure that you need to 'await' here, will verify this a bit later today